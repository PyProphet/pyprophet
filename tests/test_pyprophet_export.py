from __future__ import print_function

import os
import subprocess
import shutil
import sys

import pandas as pd
import sqlite3

import pytest

pd.options.display.expand_frame_repr = False
pd.options.display.precision = 4
pd.options.display.max_columns = None

DATA_FOLDER = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")

def _run_cmdline(cmdline):
    stdout = cmdline + "\n"
    try:
        stdout += str(subprocess.check_output(cmdline, shell=True,
                                          stderr=subprocess.STDOUT))
    except subprocess.CalledProcessError as error:
        print(error, end="", file=sys.stderr)
        raise
    return stdout


def _run_osw(regtest, temp_folder, transition_quantification=False, peptide=False, protein=False):
    os.chdir(temp_folder)
    data_path = os.path.join(DATA_FOLDER, "test_data.osw")
    shutil.copy(data_path, temp_folder)
    # MS1-level
    cmdline = "pyprophet score --in=test_data.osw --level=ms2 --test --pi0_lambda=0.001 0 0 --ss_iteration_fdr=0.02"

    # peptide-level
    cmdline += " peptide --pi0_lambda=0.001 0 0 --in=test_data.osw --context=run-specific"
    cmdline += " peptide --pi0_lambda=0.001 0 0 --in=test_data.osw --context=experiment-wide"
    cmdline += " peptide --pi0_lambda=0.001 0 0 --in=test_data.osw --context=global"

    # protein-level
    cmdline += " protein --pi0_lambda=0 0 0 --in=test_data.osw --context=run-specific"
    cmdline += " protein --pi0_lambda=0 0 0 --in=test_data.osw --context=experiment-wide"
    cmdline += " protein --pi0_lambda=0 0 0 --in=test_data.osw --context=global"

    # export
    cmdline += " export --in=test_data.osw --max_rs_peakgroup_qvalue=1 --format=legacy_merged"

    if not transition_quantification:
        cmdline += " --no-transition_quantification"

    if not peptide:
        cmdline += " --no-peptide"

    if not protein:
        cmdline += " --no-protein"

    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)

def _run_osw_unscored(regtest, temp_folder, transition_quantification=False, peptide=False, protein=False):
    os.chdir(temp_folder)
    data_path = os.path.join(DATA_FOLDER, "test_data.osw")
    shutil.copy(data_path, temp_folder)

    # export
    cmdline = "pyprophet export --in=test_data.osw --out=test_data.tsv --format=legacy_merged"

    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)

def _run_ipf(regtest, temp_folder, transition_quantification=False, ipf="disable"):
    os.chdir(temp_folder)
    data_path = os.path.join(DATA_FOLDER, "test_data.osw")
    shutil.copy(data_path, temp_folder)
    # MS1-level
    cmdline = "pyprophet score --in=test_data.osw --level=ms1 --test --pi0_lambda=0.1 0 0 --ss_iteration_fdr=0.02"

    # MS2-level
    cmdline += " score --in=test_data.osw --level=ms2 --test --pi0_lambda=0.001 0 0 --ss_iteration_fdr=0.02"

    # transition-level
    cmdline += " score --in=test_data.osw --level=transition --test --pi0_lambda=0.1 0 0 --ss_iteration_fdr=0.02"

    # IPF
    cmdline += " ipf --in=test_data.osw"

    # export
    cmdline += " export --in=test_data.osw --no-peptide --no-protein --ipf_max_peptidoform_pep=1  --max_rs_peakgroup_qvalue=1 --format=legacy_merged"

    if not transition_quantification:
        cmdline += " --no-transition_quantification"

    cmdline += " --ipf=" + ipf

    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)
    
def _run_compound_unscored(regtest, temp_folder):
    os.chdir(temp_folder)
    data_path= os.path.join(DATA_FOLDER, "test_data_compound.osw")
    shutil.copy(data_path, temp_folder)
    
    # export
    cmdline = "pyprophet export-compound --in=test_data_compound.osw --out=test_data_compound_unscored.tsv --format=legacy_merged"
    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data_compound_unscored.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)

def _run_compound_ms1(regtest, temp_folder):
    os.chdir(temp_folder)
    data_path= os.path.join(DATA_FOLDER, "test_data_compound.osw")
    shutil.copy(data_path, temp_folder)
    
    # MS1-level
    cmdline = "pyprophet score --in=test_data_compound.osw --level=ms1 --test"

    # export
    cmdline += " export-compound --in=test_data_compound.osw --out=test_data_compound_ms1.tsv --max_rs_peakgroup_qvalue=0.05 --format=legacy_merged"
    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data_compound_ms1.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)
    
def _run_compound_ms2(regtest, temp_folder):
    os.chdir(temp_folder)
    data_path= os.path.join(DATA_FOLDER, "test_data_compound.osw")
    shutil.copy(data_path, temp_folder)
    
    # MS2-level
    cmdline = "pyprophet score --in=test_data_compound.osw --level=ms2 --test"

    # export
    cmdline += " export-compound --in=test_data_compound.osw --out=test_data_compound_ms2.tsv --max_rs_peakgroup_qvalue=0.05 --format=legacy_merged"

    stdout = _run_cmdline(cmdline)

    print(pd.read_csv("test_data_compound_ms2.tsv", sep="\t", nrows=100).sort_index(axis=1),file=regtest)


def test_osw_0(tmpdir, regtest):
    _run_osw(regtest, tmpdir.strpath, False, False, False)

def test_osw_1(tmpdir, regtest):
    _run_osw(regtest, tmpdir.strpath, True, False, False)

def test_osw_2(tmpdir, regtest):
    _run_osw(regtest, tmpdir.strpath, False, True, False)

def test_osw_3(tmpdir, regtest):
    _run_osw(regtest, tmpdir.strpath, False, False, True)

def test_ipf_0(tmpdir, regtest):
    _run_ipf(regtest, tmpdir.strpath, False, "disable")

def test_ipf_1(tmpdir, regtest):
    _run_ipf(regtest, tmpdir.strpath, True, "disable")

def test_ipf_2(tmpdir, regtest):
    _run_ipf(regtest, tmpdir.strpath, False, "peptidoform")

def test_ipf_3(tmpdir, regtest):
    _run_ipf(regtest, tmpdir.strpath, False, "augmented")

def test_compound_0(tmpdir, regtest):
    _run_compound_ms1(regtest, tmpdir.strpath)
    
def test_compound_1(tmpdir, regtest):
    _run_compound_ms2(regtest, tmpdir.strpath)

def test_compound_unscored(tmpdir, regtest):
	_run_compound_unscored(regtest, tmpdir.strpath)
	
def test_osw_unscored(tmpdir, regtest):
	_run_osw_unscored(regtest, tmpdir.strpath)
