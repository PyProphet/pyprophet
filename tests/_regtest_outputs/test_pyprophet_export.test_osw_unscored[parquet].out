    Charge                  FullPeptideName  Intensity                                        ProteinName       RT             Sequence                           aggr_Fragment_Annotation                                     aggr_Peak_Apex                                     aggr_Peak_Area  aggr_prec_Peak_Apex  aggr_prec_Peak_Area  assay_iRT   assay_rt  d_score  decoy  delta_iRT  delta_rt                         filename      iRT                   id  leftWidth  m_score        mz  peak_group_rank  rightWidth               run_id  transition_group_id
0        2       ADSTGTLVITDPTR(UniMod:267)   207283.0  AQUA4SWATH_HMLangeA;AQUA4SWATH_HMLangeA;AQUA4S...  2661.55       ADSTGTLVITDPTR  1343_b4_1;1359_y3_1;1344_b5_1;1360_y4_1;1369_y...  969.0;36907.0;1426.0;6131.0;416.0;2071.0;11984...  10322.0;251772.0;9915.0;43365.0;5402.0;15040.0...           117220.750             854645.0       26.5  2595.5788   5.7301  False     1.9379   65.9712  napedro_L120420_010_SW.mzXML.gz  28.4379 -4409520928686189639  2640.5100   0.0033  728.8795              1.0   2705.3701 -8670811102654834151                    0
1        2        ALGYEDATQALGR(UniMod:267)   356015.0  AQUA4SWATH_HMLangeA;AQUA4SWATH_HMLangeA;AQUA4S...  2915.46        ALGYEDATQALGR  1381_b7_2;1397_y7_2;1383_b8_2;1399_y8_2;1385_b...  20.0;30.0;612.0;305.0;426.0;407.0;0.0;6434.0;3...  371.0;190.0;11442.0;6748.0;6892.0;3100.0;490.0...           184097.500            1350180.0       36.8  2949.3521   5.6547  False    -0.9997  -33.8921  napedro_L120420_010_SW.mzXML.gz  35.8003 -7771919224870429764  2892.9299   0.0033  687.8480              1.0   2947.5500 -8670811102654834151                    1
2        2        APIPTALDTDSSK(UniMod:259)   446618.0  AQUA4SWATH_HMLangeA;AQUA4SWATH_HMLangeA;AQUA4S...  2401.28        APIPTALDTDSSK  1427_y7_2;1429_y8_2;1416_b9_2;1431_y9_2;1406_b...  632.0;61.0;50.0;815.0;30.0;215.0;1908.0;4851.0...  14328.0;411.0;521.0;10587.0;1006.0;1832.0;1500...           384029.060            1982920.0       19.1  2341.4023   5.4280  False     1.7913   59.8777  napedro_L120420_010_SW.mzXML.gz  20.8913  -797725006165535344  2384.1599   0.0033  662.3477              1.0   2428.5400 -8670811102654834151                    2
3        2            DAVNDITAK(UniMod:259)   550550.0  AQUA4SWATH_HMLangeA;AQUA4SWATH_HMLangeA;AQUA4S...  1440.97            DAVNDITAK  1437_b7_2;1439_b8_2;1446_y8_2;1432_-1_1;1444_y...  928.0;264.0;1364.0;20899.0;8915.0;2857.0;9466....  9951.0;3196.0;9340.0;159321.0;48372.0;21445.0;...           346652.280            1718980.0       -7.8  1417.4678   5.6703  False     0.8459   23.5022  napedro_L120420_010_SW.mzXML.gz  -6.9541  8126738070040012582  1420.7200   0.0033  477.7527              1.0   1461.6899 -8670811102654834151                    3
4        2          DITAFDETLFR(UniMod:267)   316323.0  AQUA4SWATH_HMLangeA;AQUA4SWATH_HMLangeA;AQUA4S...  4736.91          DITAFDETLFR  1465_y6_2;1454_b7_2;1458_b9_2;1469_y8_2;1449_b...  30.0;10.0;60.0;101.0;397.0;100.0;1395.0;554.0;...  250.0;170.0;903.0;2148.0;6455.0;1621.0;15424.0...            89199.610             569321.0       88.1  4711.3648   5.8401  False     0.5151   25.5452  napedro_L120420_010_SW.mzXML.gz  88.6151 -1732939685941081620  4715.8398   0.0033  669.3342              1.0   4770.4702 -8670811102654834151                    4
..     ...                              ...        ...                                                ...      ...                  ...                                                ...                                                ...                                                ...                  ...                  ...        ...        ...      ...    ...        ...       ...                              ...      ...                  ...        ...      ...       ...              ...         ...                  ...                  ...
95       2           TFADGSLAFR(UniMod:267)   788463.0  AQUA4SWATH_HMLangeC;AQUA4SWATH_HMLangeC;AQUA4S...  3138.11           TFADGSLAFR  3148_b8_2;3156_y7_2;3158_y8_2;3150_b9_2;3153_y...  808.0;663.0;672.0;110.0;5583.0;2776.0;8338.0;4...  10893.0;16628.0;5533.0;1810.0;38118.0;22750.0;...           264521.100            2012060.0       44.2  3203.5231   5.3166  False    -1.9437  -65.4131  napedro_L120420_010_SW.mzXML.gz  42.2563 -5006126068043441226  3117.6399   0.0033  547.7789              1.0   3168.8401 -8670811102654834151                   65
96       2           TFADGSLAFR(UniMod:267)   137139.0  AQUA4SWATH_HMLangeC;AQUA4SWATH_HMLangeC;AQUA4S...  3010.68           TFADGSLAFR  3148_b8_2;3156_y7_2;3151_y3_1;3158_y8_2;3143_b...  880.0;2427.0;1041.0;426.0;8547.0;161.0;4623.0;...  19119.0;24871.0;13958.0;4986.0;48658.0;2301.0;...            24208.418             280044.0       44.2  3203.5268   2.1156  False    -5.6388 -192.8468  napedro_L120420_010_SW.mzXML.gz  38.5612 -3923479974150751688  2974.2500   0.0210  547.7789              2.0   3035.7000 -8670811102654834151                   65
97       2  TSDTGEEEAITDSTEANGK(UniMod:259)    38519.0  AQUA4SWATH_HMLangeC;AQUA4SWATH_HMLangeC;AQUA4S...  1857.78  TSDTGEEEAITDSTEANGK  3211_y7_2;3184_b7_2;3213_y8_2;3186_b8_2;3188_b...  0.0;10.0;233.0;130.0;0.0;0.0;183.0;40.0;0.0;10...  30.0;10.0;1566.0;811.0;20.0;10.0;836.0;544.0;4...            42908.180             195600.0        6.1  1894.8893   5.7596  False    -0.9681  -37.1093  napedro_L120420_010_SW.mzXML.gz   5.1319  5044606440762346646  1842.6700   0.0033  981.9287              1.0   1876.8101 -8670811102654834151                   66
98       2          ADHSGAVVLLK(UniMod:259)    37110.0  AQUA4SWATH_HMLangeD;AQUA4SWATH_HMLangeD;AQUA4S...  1973.12          ADHSGAVVLLK  3229_y3_1;3238_y9_2;3220_b5_1;3230_y4_1;3216_-...  888.0;2833.0;2009.0;214.0;17756.0;2603.0;1101....  6918.0;27993.0;19438.0;2753.0;195730.0;24406.0...            75249.445             897889.0        9.9  2025.4137   4.1353  False    -1.4238  -52.2937  napedro_L120420_010_SW.mzXML.gz   8.4762  1099848936678289217  1943.3500   0.0062  559.3264              1.0   1997.9800 -8670811102654834151                   67
99       2         EDNTGVVSQTFK(UniMod:259)   550945.0  AQUA4SWATH_HMLangeD;AQUA4SWATH_HMLangeD;AQUA4S...  2070.68         EDNTGVVSQTFK  3249_b7_2;3262_y6_2;3251_b8_2;3264_y7_2;3266_y...  858.0;1086.0;30.0;122.0;307.0;20.0;518.0;571.0...  7941.0;9017.0;312.0;422.0;2683.0;180.0;1936.0;...           395820.470            1666220.0       10.8  2056.3210   5.8824  False     0.5052   14.3590  napedro_L120420_010_SW.mzXML.gz  11.3052 -4744925383774838539  2053.0100   0.0033  666.8297              1.0   2087.1499 -8670811102654834151                   68

[100 rows x 27 columns]
